DiffusionPreprocessing/scripts/basic_preproc_best_b0.sh

```diff
@@ -0,0 +1,292 @@
+
+		# Select sub-set of other B0's to run topup on; we use the first ${N_other}
+		${FSLDIR}/bin/fslroi ${rawdir}/all_${pe_other}_b0s ${select_b0_dir}/opposite_b0s 0 ${N_other}
+
+		# Merge all b0's to do a rough initial aligment using topup
+		${FSLDIR}/bin/fslmerge -t ${select_b0_dir}/all_b0s ${rawdir}/all_${pe_sign}_b0s ${select_b0_dir}/opposite_b0s
+
+		# Create the acqparams file for the initial alignment
+		for idx in $(seq 1 ${N_b0s}); do
+			if [ ${PEdir} -eq 1 ]; then #RL/LR phase encoding
+				echo 1 0 0 ${ro_time} >>${select_b0_dir}/acqparams.txt
+			elif [ ${PEdir} -eq 2 ]; then #AP/PA phase encoding
+				echo 0 1 0 ${ro_time} >>${select_b0_dir}/acqparams.txt
+			fi
+		done
+
+		for idx in $(seq 1 ${N_other}); do
+			if [ ${PEdir} -eq 1 ]; then #RL/LR phase encoding
+				echo -1 0 0 ${ro_time} >>${select_b0_dir}/acqparams.txt
+			elif [ ${PEdir} -eq 2 ]; then #AP/PA phase encoding
+				echo 0 -1 0 ${ro_time} >>${select_b0_dir}/acqparams.txt
+			fi
+		done
+
+		dimz=$(${FSLDIR}/bin/fslval ${select_b0_dir}/all_b0s dim3)
+		if [ $(isodd $dimz) -eq 1 ]; then
+			${FSLDIR}/bin/fslroi ${select_b0_dir}/all_b0s ${select_b0_dir}/all_b0s_even 0 -1 0 -1 1 -1
+		else
+			${FSLDIR}/bin/imcp ${select_b0_dir}/all_b0s ${select_b0_dir}/all_b0s_even
+		fi
+		# run topup to roughly align the b0's
+		configdir=${HCPPIPEDIR_Config}
+		topup_config_file=${configdir}/best_b0.cnf
+		${FSLDIR}/bin/topup --imain=${select_b0_dir}/all_b0s_even \
+			--datain=${select_b0_dir}/acqparams.txt \
+			--config=${topup_config_file} \
+			--fout=${select_b0_dir}/fieldmap \
+			--iout=${select_b0_dir}/topup_b0s \
+			--out=${select_b0_dir}/topup_results \
+			-v
+
+		# compute squared residual from the mean b0
+		# once "bad" b0's are identified, the mean b0 will be recomputed without them
+		# and the scores recomputed; this process is iterated 3 times
+		${FSLDIR}/bin/fslmaths ${select_b0_dir}/topup_b0s -Tmean ${select_b0_dir}/topup_b0s_avg
+		for ((i = 1; i <= 3; i++)); do
+			${FSLDIR}/bin/fslmaths ${select_b0_dir}/topup_b0s -sub ${select_b0_dir}/topup_b0s_avg -sqr ${select_b0_dir}/topup_b0s_res
+
+			# Get brain mask from averaged results
+			${FSLDIR}/bin/bet ${select_b0_dir}/topup_b0s_avg ${select_b0_dir}/nodif_brain -m -R -f 0.3
+
+			# compute average squared residual over brain mask
+			scores=($(${FSLDIR}/bin/fslstats -t ${select_b0_dir}/topup_b0s_res -k ${select_b0_dir}/nodif_brain_mask -M))
+			scores_str="${scores[@]}"
+
+			echo "Iteration ${i} in finding best b0 for ${pe_sign}"
+			echo "Current scores: ${scores_str}"
+
+			# Recomputes the average using only the b0's with scores below (median(score) + 2 * mad(score)),
+			# where mad is the median absolute deviation.
+			idx=$(fslpython -c "from numpy import median, where; sc = [float(s) for s in '${scores_str}'.split()]; print(','.join(str(idx) for idx in where(sc < median(abs(sc - median(sc)) * 2 + median(sc)))[0]))")
+			echo "Recomputing average b0 using indices: ${idx}"
```

Comment:
Is this particular index counting from zero? If so, let's say that:
```suggestion
			echo "Recomputing average b0 using indices (counting from zero): ${idx}"
```